cronback-services/src/dispatcher/attempt_store.rs

```diff
@@ -1,60 +1,43 @@
-use async_trait::async_trait;
 use lib::prelude::*;
+#[cfg(test)]
 use proto::common::PaginationIn;
-use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, QueryFilter};
-
-use super::db_model::{attempts, Attempt, Attempts};
-
-pub type AttemptLogStoreError = DatabaseError;
-
-#[async_trait]
-pub trait AttemptLogStore {
-    async fn log_attempt(
-        &self,
-        attempt: Attempt,
-    ) -> Result<(), AttemptLogStoreError>;
+use sea_orm::ActiveModelTrait;
+#[cfg(test)]
+use sea_orm::{ColumnTrait, EntityTrait, QueryFilter};
 
-    async fn get_attempts_for_run(
-        &self,
-        project: &ValidShardedId<ProjectId>,
-        id: &RunId,
-        pagination: PaginationIn,
-    ) -> Result<PaginatedResponse<Attempt>, AttemptLogStoreError>;
+#[cfg(test)]
+use super::db_model::Attempts;
+use super::db_model::{attempts, Attempt};
 
-    async fn get_attempt(
-        &self,
-        project: &ValidShardedId<ProjectId>,
-        id: &AttemptId,
-    ) -> Result<Option<Attempt>, AttemptLogStoreError>;
-}
+pub type AttemptStoreError = DatabaseError;
 
-pub struct SqlAttemptLogStore {
+#[derive(Clone)]
+pub struct AttemptStore {
     db: Database,
 }
 
-impl SqlAttemptLogStore {
+impl AttemptStore {
     pub fn new(db: Database) -> Self {
         Self { db }
     }
-}
 
-#[async_trait]
-impl AttemptLogStore for SqlAttemptLogStore {
-    async fn log_attempt(
+    pub async fn log_attempt(
         &self,
         attempt: Attempt,
-    ) -> Result<(), AttemptLogStoreError> {
+    ) -> Result<(), AttemptStoreError> {
         let active_model: attempts::ActiveModel = attempt.into();
         active_model.insert(&self.db.orm).await?;
         Ok(())
     }
 
-    async fn get_attempts_for_run(
+    // Leaving this under cfg(test) until someone actually need it.
+    #[cfg(test)]
```

Comment:
I understand that this is currently unused, but I don't see the value of marking it as test tbh

Reply:
clippy